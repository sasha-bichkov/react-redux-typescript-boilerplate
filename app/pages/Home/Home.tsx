import React from 'react'
import { withTranslation, TFunction } from 'react-i18next'

import OmniAuthButton from '@Components/OmniAuthButton'
import GoogleIcon from '@Images/OmniAuthButton/GoogleLogo.svg'

import './Home.scss'

interface HomeProps {
  t: TFunction<('translation' | 'common')[], undefined>
}

interface HomeState {} // eslint-disable-line @typescript-eslint/no-empty-interface

class Home extends React.Component<HomeProps, HomeState> {
  render() {
    const { t } = this.props

    return (
      <div>
        <h1>{t('home.title')}</h1>

        <OmniAuthButton Icon={GoogleIcon} onClick={() => console.log('click')}>
          Sign in with Google
        </OmniAuthButton>
      </div>
    )
  }
}

export default withTranslation()(Home)
