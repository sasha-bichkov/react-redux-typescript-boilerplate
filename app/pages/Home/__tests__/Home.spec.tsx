import React from 'react'
import { shallow } from 'enzyme'
import Home from '../Home'

describe('Home', () => {
  it('renders correctly', () => {
    const component = shallow(<Home />)

    expect(component).toMatchSnapshot()
  })
})
