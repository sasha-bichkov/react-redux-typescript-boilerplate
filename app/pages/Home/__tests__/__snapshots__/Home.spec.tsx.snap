// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <UnsupportedBrowser />
</div>
`;

exports[`Home should render correctly in "debug" mode 1`] = `
<div>
  <UnsupportedBrowser />
</div>
`;
